metrics-httpclient/src/main/java/com/codahale/metrics/httpclient/InstrumentedHttpClientConnectionManager.java

```diff
@@ -114,4 +82,76 @@ public void shutdown() {
         metricsRegistry.remove(name(HttpClientConnectionManager.class, name, "max-connections"));
         metricsRegistry.remove(name(HttpClientConnectionManager.class, name, "pending-connections"));
     }
+
+    public static Builder builder(final MetricRegistry metricsRegistry) {
+        return new Builder().metricsRegistry(metricsRegistry);
+    }
+
+    public static class Builder {
+        private MetricRegistry metricsRegistry;
+        private HttpClientConnectionOperator httpClientConnectionOperator;
+        private Registry<ConnectionSocketFactory> socketFactoryRegistry = getDefaultRegistry();
+        private HttpConnectionFactory<HttpRoute, ManagedHttpClientConnection> connFactory;
+        private SchemePortResolver schemePortResolver;
+        private DnsResolver dnsResolver = SystemDefaultDnsResolver.INSTANCE;
+        private long connTTL = -1;
+        private TimeUnit connTTLTimeUnit = TimeUnit.MILLISECONDS;
+        private String name;
+
+        Builder() {
+        }
+
+        public Builder metricsRegistry(final MetricRegistry metricsRegistry) {
```

Comment:
To be honest, I don't see much value in declaring the method parameters as `final`. It's clear from the semantic of `Builder` that they are not going to change and we don't force final method parameters in the codestyle. I would remove them unless there's a specific reason to keep.

Reply:
There are two notes from my side here:
1) I think these improve readability a bit, though I might be mistaken
2) TBH, this is an autogenerated code. I first did it with lombok and then delombokize the code to get the template out.

Again - since this is your repo, not mine, will remove "final", no problem.